### PR TITLE
refactor: add lazy field to LibraryCollectionData

### PR DESCRIPTION
**Pull request by navinkarkera**
This is required to handle the signal asynchronously whenever the sender does not care about the handler execution completion.

For example, whenever multiple collections are updated due to a component being added or removed, we do not want to wait for all collection update handlers to complete.

**Dependencies:**

* https://github.com/openedx/edx-platform/pull/35600
* https://github.com/openedx/openedx-learning/pull/238
* https://github.com/openedx/frontend-app-authoring/pull/1373

**Merge deadline:** ASAP

**Testing instructions:**

* See https://github.com/openedx/frontend-app-authoring/pull/1373 for test instructions.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @navinkarkera!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/hooks-extension-framework`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

**Review by rpenido — APPROVED** [review]
LGTM :+1:
Thank you for your work, @navinkarkera!

- [x] I tested this using the instructions from: https://github.com/openedx/frontend-app-authoring/pull/1373
- [x] I read through the code
- [x] ~I checked for accessibility issues~ N/A
- [x] Includes documentation

**Comment by ChrisChV** [comment]
@felipemontoya @mariajgrimaldi @brian-smith-tcril Could you review this as CC? Thanks!

File: openedx_events/content_authoring/data.py
Comment (mariajgrimaldi) [inline openedx_events/content_authoring/data.py:225]: Please, don't consider this comment a blocker. I’m having a bit of trouble understanding what lazy means in this context without reading the description. Do you imagine other cases where this field is `True` besides bulk updates? Should the name suggest that?

Comment (navinkarkera) [reply openedx_events/content_authoring/data.py:225]: @mariajgrimaldi We want to control the way handler executes some functions (search indexing in this case). How about renaming this field to `async` or `background`. 
cc: @rpenido

Comment (mariajgrimaldi) [reply openedx_events/content_authoring/data.py:225]: Yes. That’d be more self-explanatory. Thank you!

Comment (navinkarkera) [reply openedx_events/content_authoring/data.py]: Done.

**Review by mariajgrimaldi — APPROVED** [review]
Thank you for addressing my comments! Feel free to merge and release when you see fit.

**Review by ChrisChV — APPROVED** [review]
Looks good! :+1:

**Comment by bradenmacdonald** [comment]
@mariajgrimaldi Would you mind merging for us? I don't think we have permission.

**Comment by mariajgrimaldi** [comment]
@bradenmacdonald: @navinkarkera has merge rights. But yes, I’ll merge as soon as I can.

**Comment by navinkarkera** [comment]
@mariajgrimaldi Thank you!

**Pull request merged**
